src/main/java/ch/njol/skript/effects/EffLog.java

```diff
@@ -122,13 +140,13 @@ protected void execute(final Event e) {
 					if (script != null)
 						scriptName = script.getConfig().getFileName();
 				}
-				Skript.info("[" + scriptName + "] " + message);
+				SkriptLogger.LOGGER.log(logLevel, "[" + scriptName + "] " + messages);
```

Comment:
```suggestion
				SkriptLogger.LOGGER.log(logLevel, "[" + scriptName + "] " + message);
```
`messages` is the expression field, right?

Reply:
`messages = (Expression<String>) exprs[0];` basically, messages is the text the players put to log something.
`files = (Expression<String>) exprs[1];` files means the file they chose to log it into (or if its not specified, it defaults to server.log)

Reply:
right. it looks like you are accidentally logging the toString of the messages expression rather than the actual message (hence the suggested change)